### PR TITLE
don't click when disabled

### LINKED ISSUE #383: click helper should ignore disabled inputs
`this.$('#foo').click()` and `document.getElementById('foo').click()` do nothing if the input is disabled. `await click('#foo')` still clicks disabled inputs. This is probably a bug.

### PR DESCRIPTION
**Pull request by kellyselden**
Closes #383.

Probably affects other helpers.

### CODE REVIEW

File: addon-test-support/@ember/test-helpers/dom/click.js [hunk 1]
```diff
@@ -59,7 +59,10 @@ export default function click(target) {
       throw new Error(`Element not found when calling \`click('${target}')\`.`);
     }
 
-    __click__(element);
+    if (!element.hasAttribute('disabled')) {
```
Comment (Turbo87) [inline addon-test-support/@ember/test-helpers/dom/click.js]: I think it should check the property instead of the attribute 🤔

Comment (kellyselden) [reply addon-test-support/@ember/test-helpers/dom/click.js]: updated

**Comment by cibernox** [comment]
I'm confused. I understand that disabled inputs can't be filled, but can't they be clicked? That's surprising

**Comment by Turbo87** [comment]
@cibernox I think you're confusing `disabled` with `readonly`

**Comment by cibernox** [comment]
I'm not. I only expected disabled inputs to not be fillable or focusable, I didn't know they weren't clickable

File: addon-test-support/@ember/test-helpers/dom/click.js [hunk 2]
```diff
@@ -59,7 +59,10 @@ export default function click(target) {
       throw new Error(`Element not found when calling \`click('${target}')\`.`);
     }
 
-    __click__(element);
+    if (!element.disabled) {
```
[the remark this answers is not in the document]
Comment (kellyselden) [reply addon-test-support/@ember/test-helpers/dom/click.js]: updated

**Comment by rwjblue** [comment]
Thank you @kellyselden!

**Pull request merged**